src/actions/action.js

```diff
@@ -0,0 +1,45 @@
+import { POST_ACTION_SUCCEEDED } from '../types';
+import { POST_ACTION_CHANNEL } from '../config/channels';
+
+const postAction = async (
+  {
+    userId,
+    id: appInstanceId,
+    spaceId,
+    subSpaceId,
+    format,
+    data,
+    verb,
+    geolocation,
+    visibility,
+  } = {},
+  callback
+) => () => {
+  try {
+    window.ipcRenderer.send(POST_ACTION_CHANNEL, {
+      userId,
+      appInstanceId,
+      spaceId,
+      subSpaceId,
+      format,
+      data,
+      verb,
+      geolocation,
+      visibility,
+    });
+
+    window.ipcRenderer.once(POST_ACTION_CHANNEL, async (event, response) => {
+      callback({
+        // have to include the appInstanceId to avoid broadcasting
+        appInstanceId,
+        type: POST_ACTION_SUCCEEDED,
+        payload: response,
+      });
+    });
+  } catch (err) {
+    console.error(err);
```

Comment:
Here we can't use `Logger`, because we're in the front end. However, eventually, we should not be using logging the console error if it's not something we want to show to the user. If we want to show something to the user, we should use a toast. If we don't, then we can just replace this by `//  do nothing` for now. We could also use Sentry to send us the log of the error, but if it's something that is just caused by lack of internet or whatever, it's better not to pollute our error logs.